pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyAnonymousRoleTest.java

```diff
@@ -0,0 +1,151 @@
+        proxyConfig.getProperties().setProperty("tokenSecretKey", "data:;base64," + Base64.getEncoder().encodeToString(SECRET_KEY.getEncoded()));
+        proxyConfig.setBrokerServiceURL(pulsar.getBrokerServiceUrl());
+        proxyConfig.setServicePort(Optional.of(0));
+        proxyConfig.setBrokerProxyAllowedTargetPorts("*");
+        proxyConfig.setWebServicePort(Optional.of(0));
+        proxyConfig.setAuthenticationProviders(providers);
+
+        proxyService = Mockito.spy(new ProxyService(proxyConfig,
+                new AuthenticationService(
+                        PulsarConfigurationLoader.convertFrom(proxyConfig))));
+    }
+
+    @AfterMethod(alwaysRun = true)
+    @Override
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+        proxyService.close();
+    }
+
+    private void startProxy() throws Exception {
+        proxyService.start();
+    }
+
+    @Test
+    public void testAuthorizationWithProxyAsAnonymousRoleWithOriginalRole() throws Exception {
+        log.info("-- Starting {} test --", methodName);
+
+        startProxy();
+        createAdminClient();
+
+        String namespaceName = "my-property/proxy-authorization/my-ns";
+        admin.clusters().createCluster("proxy-authorization", ClusterData.builder().serviceUrl(brokerUrl.toString()).build());
+        admin.tenants().createTenant("my-property",
+                new TenantInfoImpl(Sets.newHashSet("appid1", "appid2"), Sets.newHashSet("proxy-authorization")));
+        admin.namespaces().createNamespace(namespaceName);
+
+        @Cleanup
+        PulsarClient proxyClient = createPulsarClient(proxyService.getServiceUrl(), PulsarClient.builder());
+
+        // client role does not have permission to consume to a topic
+        assertThatThrownBy(() -> proxyClient.newConsumer()
+                .topic("persistent://my-property/proxy-authorization/my-ns/my-topic1")
+                .subscriptionName("my-subscriber-name").subscribe());
+
+        // grant consume permission to client role
+        admin.namespaces().grantPermissionOnNamespace(namespaceName, CLIENT_ROLE,
+                Sets.newHashSet(AuthAction.consume));
+        proxyClient.newConsumer()
+                .topic("persistent://my-property/proxy-authorization/my-ns/my-topic1")
+                .subscriptionName("my-subscriber-name").subscribe();
+    }
+    private void createAdminClient() throws Exception {
+        admin = spy(PulsarAdmin.builder().serviceHttpUrl(brokerUrl.toString())
+                .authentication(AuthenticationFactory.token(ADMIN_TOKEN)).build());
+    }
+
+    private PulsarClient createPulsarClient(String proxyServiceUrl, ClientBuilder clientBuilder)
+            throws PulsarClientException {
+        return clientBuilder.serviceUrl(proxyServiceUrl).statsInterval(0, TimeUnit.SECONDS)
+                .authentication(AuthenticationFactory.token(CLIENT_TOKEN))
+                .operationTimeout(1000, TimeUnit.MILLISECONDS).build();
```

Comment:
Why do we set a timeout here?
It is better to not set parameters if not needed.
I'm this case 1 second may cause the test di be flaky

Reply:
No reason, I can remove this.